enterprise/coderd/groups.go

```diff
@@ -102,36 +102,59 @@ func (api *API) patchGroup(rw http.ResponseWriter, r *http.Request) {
 	)
 	defer commitAudit()
 
-	currentMembers, currentMembersErr := api.Database.GetGroupMembers(ctx, group.ID)
-	if currentMembersErr != nil {
-		httpapi.InternalServerError(rw, currentMembersErr)
+	var req codersdk.PatchGroupRequest
+	if !httpapi.Read(ctx, rw, r, &req) {
 		return
 	}
 
-	aReq.Old = group.Auditable(currentMembers)
+	// If the name matches the existing group name pretend we aren't
+	// updating the name at all.
+	if req.Name == group.Name {
+		req.Name = ""
+	}
 
-	var req codersdk.PatchGroupRequest
-	if !httpapi.Read(ctx, rw, r, &req) {
+	if group.ID == group.OrganizationID && req.Name != "" {
+		httpapi.Write(ctx, rw, http.StatusBadRequest, codersdk.Response{
+			Message: fmt.Sprintf("Cannot rename the %q group!", database.AllUsersGroup),
+		})
 		return
 	}
 
-	if req.Name != "" && req.Name == database.AllUsersGroup {
+	if group.ID == group.OrganizationID && (req.DisplayName != nil && *req.DisplayName != "") {
 		httpapi.Write(ctx, rw, http.StatusBadRequest, codersdk.Response{
-			Message: fmt.Sprintf("%q is a reserved group name!", database.AllUsersGroup),
+			Message: fmt.Sprintf("Cannot update the Display Name for the %q group!", database.AllUsersGroup),
 		})
 		return
 	}
 
-	// If the name matches the existing group name pretend we aren't
-	// updating the name at all.
-	if req.Name == group.Name {
-		req.Name = ""
+	if req.Name != "" && req.Name == database.AllUsersGroup {
+		httpapi.Write(ctx, rw, http.StatusBadRequest, codersdk.Response{
+			Message: fmt.Sprintf("%q is a reserved group name!", database.AllUsersGroup),
+		})
+		return
 	}
 
 	users := make([]string, 0, len(req.AddUsers)+len(req.RemoveUsers))
 	users = append(users, req.AddUsers...)
 	users = append(users, req.RemoveUsers...)
```

Comment:
Can we rename this to `modifiedUsers` or `diffUsers`? `users` is just a bit confusing.